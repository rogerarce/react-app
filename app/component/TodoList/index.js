import React from 'react'

class TodoList extends React.PureComponent {
  constructor(props) {
    super(props)

    this.state = {
      todos: [
        { key: 1, title: 'Testing todo 1', completed: false },
        { key: 2, title: 'Testing todo 2', completed: false },
        { key: 3, title: 'Testing todo 3', completed: false },
      ]
    }
  }

  render() {
    return (
      <ul>
        {this.state.todos.map(todo => <li key={todo.key}>{todo.title}</li>)}
      </ul>
    )
  }
}

export default TodoList
