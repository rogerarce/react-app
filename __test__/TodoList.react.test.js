/*
 *
 */
import React from 'react'
import TodoList from '../app/component/TodoList'
import renderer from 'react-test-renderer'

test('TodoList component tests', () => {
  const component = renderer.create(
    <TodoList />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
