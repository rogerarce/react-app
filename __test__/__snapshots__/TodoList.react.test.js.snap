// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component tests 1`] = `
<ul>
  <li>
    Testing todo 1
  </li>
  <li>
    Testing todo 2
  </li>
  <li>
    Testing todo 3
  </li>
</ul>
`;
